premerge/buildbot/dispatch_job.py

```diff
@@ -0,0 +1,179 @@
+"""Dispatches a job to the k8s cluster.
+
+This script takes in a commit SHA to test along with the platform, spawns a job
+to test it, and then streams the logs from the job. We read logs from the job
+every so often using the kuberntes logging API rather than directly executing
+commands inside the container and streaming the output. This is to work
+around https://github.com/kubernetes-sigs/apiserver-network-proxy/issues/748.
+"""
+
+import sys
+import logging
+import time
+import dateutil
+import datetime
+import json
+
+import kubernetes
+
+PLATFORM_TO_NAMESPACE = {"Linux": "llvm-premerge-linux-builbot"}
+LOG_SECONDS_TO_QUERY = 10
+SECONDS_QUERY_LOGS_EVERY = 5
+
+
+def start_build_linux(commit_sha: str, k8s_client) -> str:
+    """Spawns a pod to build/test LLVM at the specified SHA.
+
+    Args:
+      commit_sha: The commit SHA to build/run the tests at.
+      k8s_client: The kubernetes client instance to use for spawning the pod.
+
+    Returns:
+      A string containing the name of the pod.
+    """
+    pod_name = f"build-{commit_sha}"
+    commands = [
+        "git clone --depth 100 https://github.com/llvm/llvm-project",
+        "cd llvm-project",
+        f"git checkout ${commit_sha}",
+        "export CC=clang",
+        "export CXX=clang++",
+        './.ci/monolithic-linux.sh "bolt;clang;clang-tools-extra;flang;libclc;lld;lldb;llvm;mlir;polly" "check-bolt check-clang check-clang-cir check-clang-tools check-flang check-lld check-lldb check-llvm check-mlir check-polly" "compiler-rt;libc;libcxx;libcxxabi;libunwind" "check-compiler-rt check-libc" "check-cxx check-cxxabi check-unwind" "OFF"'
+        "echo BUILD FINISHED",
+    ]
+    pod_definition = {
+        "apiVersion": "v1",
+        "kind": "Pod",
+        "metadata": {
+            "name": pod_name,
+            "namespace": PLATFORM_TO_NAMESPACE["Linux"],
+        },
+        "spec": {
+            "containers": [
+                {
+                    "name": "build",
+                    "image": "ghcr.io/llvm/ci-ubuntu-24.04",
+                    "command": ["/bin/bash", "-c", ";".join(commands)],
+                }
+            ],
+            "restartPolicy": "Never",
+        },
+    }
+    kubernetes.utils.create_from_dict(k8s_client, pod_definition)
+    return pod_name
+
+
+def read_logs(pod_name: str, namespace: str, v1_api) -> list[str]:
+    """Reads logs from the specified pod.
+
+    Reads logs using the k8s API and returns a nicely formatted list of
+    strings.
+
+    Args:
+      pod_name: The name of the pod to read logs from.
+      namespace: The namespace the pod is in.
+      v1_api: The kubernetes API instance to use for querying logs.
+
+    Returns:
+      A list of strings representing the log lines.
+    """
+    logs = v1_api.read_namespaced_pod_log(
```

Comment:
I really do not know: is "namespaced" a typo or is it correct?

Reply:
Namespaced is correct here.

https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/CoreV1Api.md#read_namespaced_pod_log